src/math_lib/math_lib.cpp

```diff
@@ -2,8 +2,29 @@
 
 namespace MathLib
 {
-	bool isEqual(double a, double b, double tolerance)
-	{
-		return abs(a - b) <= tolerance;
-	}
+    bool isEqual(double a, double b, double tolerance)
+    {
+        return abs(a - b) <= tolerance;
+    }
+
+    template <typename T>
+    double sqrt(T a)
+    {
+        if (a == 0)
+            return 0;
+        if (a < 0)
+            throw std::invalid_argument("Cannot get the square root of a negative number");
+        double x = a;
+        double result;
+        int count = 0;
+        while (true)
+        {
+            count++;
+            result = 0.5 * (x + (n / x));
+            if (abs(result - x) < 0.00001)
```

Comment:
Good job! But please use the existing isEqual function here instead of writing the same code.